conan/internal/cache/db/table.py

```diff
@@ -19,11 +21,12 @@ def __init__(self, filename):
 
     @contextmanager
     def db_connection(self):
-        connection = sqlite3.connect(self.filename, isolation_level=None, timeout=10)
-        try:
-            yield connection
-        finally:
-            connection.close()
+        with type(self)._connection_mutex:
```

Comment:
```suggestion
        with self._connection_mutex:
```
Not necessary, usage shouldn't worry about this, creation should, I am fine trusting that it is created correctly above.